Automattic/jetpack
# CRM Automations UI: Add store and workflows reducer

Fixes https://github.com/Automattic/zero-bs-crm/issues/3181
Fixes https://github.com/Automattic/zero-bs-crm/issues/3205

## Proposed changes:

This PR accomplishes two tasks:
1. Creates a Redux store for our React apps.
2. Adds a `workflows` reducer to that store for use by the Automations UI.

The store uses `@wordpress/data` and is found at `/src/state/store`. This folder sets the pattern that future projects can use to add new reducers as needed, and the store can then be provided to multiple views so that state can be shared between them.

The workflows reducer will keep the primary state for the Automations UI. I have created an initial set of actions which will support our MVP, although more actions will need to be created for future expanded versions of the Automations module.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






This code is not called anywhere yet, so just review the code and make sure the tests are comprehensive, correct, and pass. Run them with `pnpm test` from the root crm directory.
